### PR TITLE
[WIP] Document stage contracts

### PR DESCRIPTION
**Pull request by juliocc**

### CODE REVIEW

File: fast/contracts.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  type = object({
+    folder_ids = map(string)
+    automation_resources = map(object({
+      gcs = string
+      sa  = string
+    }))
+  })
+}
+
+variable "f_networking" {
```
Comment (sruffilli) [inline fast/contracts.tf:47]: How are we grouping stuff logically?
I prefer

```
shared_vpcs = {
  prod = "self_link_a"
  dev  = "self_link_b"
}
```
over 
```
prod = {
  shared_vpc = "self_link_a"
  host_project = [...]
} 
```

...but that means that consumers will cherry pick across multiple collections (nothing wrong with that).

File: fast/contracts.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  description = "Data coming from the networking stage"
+  type = object({
+    shared_vpcs       = map(string)
+    vpc_host_projects = map(string)
```
Comment (ludoo) [inline fast/contracts.tf:51]: I would shorten whenever possible: `host_projects`

**Comment by ludoo** [comment]
What about renaming to `i_xxxx` for `interface`?

File: fast/contracts.tf (hunk 2 above)
Comment (ludoo) [reply fast/contracts.tf:51]: and even `vpcs`

File: fast/contracts.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }))
+  })
+}
+
+variable "f_networking" {
+  description = "Data coming from the networking stage."
+  type = object({
+    dns_zones     = map(string)
+    host_projects = map(string)
+    vpcs          = map(string)
```
Comment (ludoo) [inline fast/contracts.tf:52]: shall we also add subnets? it won't hurt too much and might prove useful, WDYT?

Comment (sruffilli) [reply fast/contracts.tf:52]: Makes sense. PF (and I assume other S3) don't really consume them as vars, but we might want to change that.

Comment (juliocc) [reply fast/contracts.tf:52]: added

**Comment by juliocc** [comment]
Closing this in favor of #549

**Pull request closed without merging**